compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4717,6 +4722,17 @@ private void checkTypesForRecords(BLangWaitForAllExpr waitExpr, AnalyzerData dat
         }
     }
 
+    private boolean isBinaryBitwiseOperatorExpr(BLangExpression valueExpr) {
+        if (valueExpr.getKind() == NodeKind.GROUP_EXPR) {
+            return isBinaryBitwiseOperatorExpr(((BLangGroupExpr) valueExpr).expression);
+        }
+        if (valueExpr.getKind() == NodeKind.BINARY_EXPR
```

Comment:
I would move the termination condition for top.

Reply:
Refer to https://github.com/ballerina-platform/ballerina-lang/pull/37832#discussion_r1257730501